app/controllers/annotations_overview_controller.rb

```diff
@@ -0,0 +1,54 @@
+class AnnotationsOverviewController < ApplicationController
+  layout "application_no_sidebar_with_background"
+
+  def show
+    user_annotations = Annotation.where(user_id: current_user.id)
+                                 .map { |a| extract_relevant_information(a) }
+    @annotations_by_lecture = user_annotations.group_by { |annotation| annotation[:lecture] }
+    # TODO: don't include lecture key in the hash anymore after grouping
+    # Maybe we can even gruop in the SQL query directly?
+
+    @show_students_annotations = current_user.teachable_editor_or_teacher?
+    if @show_students_annotations
+      student_annotations =
+        Annotation.where(medium_id: medium_ids_for_teacher_or_editor,
+                         visible_for_teacher: true)
+                  .map { |a| extract_relevant_information(a, is_shared: true) }
+      @student_annotations_by_lecture = student_annotations.group_by do |annotation|
+        annotation[:lecture]
+      end
+    end
+
+    render "annotations/annotations_overview"
+  end
+
+  def medium_ids_for_teacher_or_editor
+    lectures = current_user.given_lectures + current_user.edited_lectures
+    lectures.map(&:media_with_inheritance).flatten.pluck(:id)
+  end
+
+  def extract_relevant_information(annotation, is_shared: false)
```

Comment:
I disagree that this extraction is "not a controller method". I've now scoped it privately in the annotations controller and renamed it to `extract_annotation_overview_information`, but I think that what this method does is exactly what a controller is for. Here, we create the interface for the view by mapping model-specific data to a Ruby hash that will be sent as JSON objects to the frontend. Instead of `category`, we could call this field `superfield` and acceses it as `superfield` in the frontend and the modal wouldn't care. This shows that this code should reside in a more outer layer and is not core business logic (in my opinion). In clean architecture, this code would be part of the outer onion rings that can rot without affecting the business logic in deeper-located rings.

Reply:
What I meant to say was actually  "is not a controller *action*". Do we agree on this?

Reply:
Yes.